### PR TITLE
Nisse/v0.7.0 radar support

### PR DESCRIPTION
**Pull request by pd-nisse**

### CODE REVIEW

File: paralleldomain/common/dgp/v0/constants.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
     "10": AnnotationTypes.SurfaceNormals2D,
     "12": AnnotationTypes.Albedo2D,
     "13": AnnotationTypes.MaterialProperties2D,
+    "15": None,
```
Comment (phillip-thomas-pd) [inline paralleldomain/common/dgp/v0/constants.py:24]: why do we need this?

File: paralleldomain/decoding/dgp/v1/sensor_frame_decoder.py [hunk 2]
```diff
@@ -660,6 +688,140 @@ def _decode_metadata(self, sensor_name: SensorName, frame_id: FrameId) -> Dict[s
...
+            ).astype(np.float32)
+        else:
+            return None
+
+    def _decode_radar_point_cloud_rgb(self, sensor_name: SensorName, frame_id: FrameId) -> Optional[np.ndarray]:
+        return None
+
+    def _decode_radar_point_cloud_power(self, sensor_name: SensorName, frame_id: FrameId) -> Optional[np.ndarray]:
+        fields = [RadarPointFormat.POWER]
+        radar_point_cloud_format = self._decode_radar_point_cloud_format(sensor_name=sensor_name, frame_id=frame_id)
```
Comment (phillip-thomas-pd) [inline paralleldomain/decoding/dgp/v1/sensor_frame_decoder.py:744]: there is a lot of code repetition here.. lets move this into a reusable method and take fields as parameter. Also you could move _decode_radar_point_cloud_data into the if statement to save some time

File: paralleldomain/decoding/directory/frame_decoder.py [hunk 3]
```diff
@@ -2,8 +2,8 @@
 
 import numpy as np
 
-from paralleldomain.decoding.directory.sensor_frame_decoder import DirectoryCameraSensorFrameDecoder
 from paralleldomain.decoding.common import DecoderSettings
+from paralleldomain.decoding.directory.sensor_frame_decoder import DirectoryCameraSensorFrameDecoder
 from paralleldomain.decoding.frame_decoder import FrameDecoder
 from paralleldomain.decoding.sensor_frame_decoder import CameraSensorFrameDecoder, LidarSensorFrameDecoder
 from paralleldomain.model.ego import EgoPose
```
Comment (phillip-thomas-pd) [inline paralleldomain/decoding/directory/frame_decoder.py:9]: The frame_decoders of the directory, nuimages, nuscenes are not implementing the abstract radar decoding methods. You can simply return empty lists or None (depending on the datatype) there. But to make sure they dont crash at runtime you should implement those.

File: paralleldomain/model/unordered_scene.py [hunk 4]
```diff
@@ -183,10 +219,16 @@ def lidar_frames(self) -> Generator[LidarSensorFrame[TDateTime], None, None]:
         for lidar in self.lidars:
             yield from lidar.sensor_frames
 
+    @property
+    def radar_frames(self) -> Generator[RadarSensorFrame[TDateTime], None, None]:
```
Comment (phillip-thomas-pd) [inline paralleldomain/model/unordered_scene.py:223]: also add the radar_frames and radar_names property to the Dataset class

File: paralleldomain/common/dgp/v0/constants.py (hunk 1 above)
Comment (pd-nisse) [reply paralleldomain/common/dgp/v0/constants.py:24]: will be resolved with the other DGP proto PR - not worth doing anything here

File: paralleldomain/model/unordered_scene.py (hunk 4 above)
Comment (GrossmanYaad) [reply paralleldomain/model/unordered_scene.py:223]: Done

File: paralleldomain/decoding/directory/frame_decoder.py (hunk 3 above)
Comment (GrossmanYaad) [reply paralleldomain/decoding/directory/frame_decoder.py:9]: Lets talk, didnt understand

File: paralleldomain/decoding/dgp/v1/sensor_frame_decoder.py (hunk 2 above)
Comment (GrossmanYaad) [reply paralleldomain/decoding/dgp/v1/sensor_frame_decoder.py]: Done

File: paralleldomain/model/unordered_scene.py (hunk 4 above)
Comment (GrossmanYaad) [reply paralleldomain/model/unordered_scene.py:223]: Reopened for other changes

Comment (GrossmanYaad) [reply paralleldomain/model/unordered_scene.py:223]: Done

File: paralleldomain/decoding/directory/frame_decoder.py (hunk 3 above)
Comment (GrossmanYaad) [reply paralleldomain/decoding/directory/frame_decoder.py:10]: Done

**Pull request merged**